lib/services/elasticsearch.js

```diff
@@ -147,9 +147,27 @@ module.exports = function (kuzzle, options) {
    * @returns {Promise} resolve an object that contains _id
    */
   this.create = function (requestObject) {
-    var data = cleanData.call(this, requestObject);
+    var
+      deferred = q.defer(),
+      data = cleanData.call(this, requestObject);
+
+    this.client.create(data)
+      .then(function (result) {
+        if (kuzzle.indexes[data.index] === undefined) {
+          kuzzle.indexes[data.index] = [];
+        }
+
+        if (kuzzle.indexes[data.index].indexOf(data.type) === -1) {
+          kuzzle.indexes[data.index].push(data.type);
+        }
```

Comment:
These few lines seems to be repeated in all ES function: maybe they should be isolated in a dedicated private method?